src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,53 @@
+package duke;
+
+import duke.exception.DukeException;
+import duke.command.Command;
+import duke.parser.Parser;
+
+/**
+ * Represents a duke object.
+ */
+public class Duke {
+
+    private Storage storage;
+    private TaskList tasks;
+    private Ui ui;
+
+    public Duke(String filePath) {
+        ui = new Ui();
+        storage = new Storage(filePath);
+        try {
+            tasks = new TaskList(storage.load());
+        } catch (DukeException e) {
+            ui.showLoadingError();
+            tasks = new TaskList();
+        }
+    }
+
+    public static void main(String[] args) {
+
+        Duke duke = new Duke("../data/duke.txt");
```

Comment:
Hi Xueting, I thought that the file path could be passed into Storage directly in the Duke constructor (without Duke having to take in an additional argument). Perhaps, you were coming from the direction that different Duke users would have different file paths, which is a good consideration, but I felt that the storage directory isn't usually a customizable attribute since we don't need users to know where we store the temp files.